resource/labeling-app/components/text-sample/textArea.js

```diff
@@ -52,65 +62,78 @@ const TextArea = {
+        handleClickOnSelection(selectionId) {
+            return (mEvent) => {
+                this.mapAndEmit((o) => {
+                    if (selectionId === this.getSelectionId(o.tokenStart, o.tokenEnd)) {
+                        o.selected = !o.selected;
+                    } else {
+                        o.selected = (mEvent.ctrlKey || mEvent.metaKey) ? o.selected : false;
+                    }
+                })
+            }
         },
-        makeSelected(startId, endId, category, selected) {
-            const wordsIds = this.getSelectedWordsFromBoundaries(startId, endId);
+        makeSelected(range, category, selected, isPrelabel) {
+            const textAreaNodes = range.startContainer.childNodes;
+            const [tokenStart, tokenEnd] = [textAreaNodes[range.startOffset], textAreaNodes[range.endOffset - 1]].map((t) => {
+                return this.parseTokenId(t).tokenIndex
+            })
             const color = category ? category.color : UNDEFINED_COLOR;
             const colorStrTransparent = this.colorToCSS(color, 0.25);
             const colorStrOpaque = this.colorToCSS(color)
 
             const caption = category ? category.caption : UNDEFINED_CAPTION;
 
-            const selectedWords = wordsIds.map((x) => document.getElementById(x));
             const selectionWrapper = document.createElement('mark');
             selectionWrapper.classList.add('selection-wrapper');
             selected && selectionWrapper.classList.add('selected');
-            const selectionId = this.getSelectionId(startId, endId);
+            const selectionId = this.getSelectionId(tokenStart, tokenEnd);
             selectionWrapper.id = selectionId;
-            selectionWrapper.addEventListener('dblclick', () => {
-                const newEntities = this.entities.filter(
-                    (x) => selectionId !== this.getSelectionId(x.startId, x.endId));
-                this.$emit("update:entities", newEntities);
-            });
-            selectionWrapper.addEventListener('click', (mEvent) => {
-                this.mapAndEmit((o) => {
-                    if (selectionId === this.getSelectionId(o.startId, o.endId)) {
-                        o.selected = !o.selected;
-                    } else {
-                        o.selected = (mEvent.ctrlKey || mEvent.metaKey) ? o.selected : false;
-                    }
-                })
-            });
-            selectionWrapper.style.background = colorStrTransparent
-            selectedWords[0].parentNode.insertBefore(selectionWrapper, selectedWords[0]);
-            selectedWords.forEach((x) => selectionWrapper.appendChild(x));
 
+            selectionWrapper.addEventListener('dblclick', this.handleDblClickOnSelection(selectionId));
+            selectionWrapper.addEventListener('click', this.handleClickOnSelection(selectionId));
+
+            if (!isPrelabel) selectionWrapper.style.background = colorStrTransparent;
```

Comment:
same comment as above, it's cleaner to handle styles in css and just assign classes based on condition

Reply:
Same comment as above, as it changes dynamically, I don't know if I can do otherwise

Reply:
my bad, didn't see that the value was dynamic 🙈

Reply:
yep, my bad, forgot about the colors 🙈